salt/states/saltmod.py

```diff
@@ -284,6 +284,8 @@ def state(name,
 
     cmd_kw['tgt_type'] = tgt_type
     cmd_kw['ssh'] = ssh
+    if 'roster' in kwargs:
+        cmd_kw['roster_file'] = kwargs['roster']
```

Comment:
`roster` argument refers to roster-system, not roster-file. "Roster-file" is relevant within to "flat" roster system, but Salt allows other roster-systems to be used including custom ones.

Reply:
The documentation says that `roster` for the orchestration state refers to a roster file. This PR is just to make the state honor what the documentation says. Passing the `roster_file` keyword arg to `saltutil.cmd` uses the default `roster` value of `flat`

Reply:
> The documentation says that roster for the orchestration state refers to a roster file.

Not it's not:

> roster
> In the event of using salt-ssh, a **roster system** can be set

https://docs.saltstack.com/en/latest/ref/states/all/salt.states.saltmod.html#salt.states.saltmod.state
Notice - roster **system**, not roster **file**

Reply:
Okay, thanks for pointing that out. I'll look at this again.

Reply:
@AndreiPashkin Can you reply to #51158 with the expected behavior for this option? I will try to re-implement this based on your feedback in the ticket.

Reply:
@AndreiPashkin Are we good here?